src/Compiler/Microsoft.AspNetCore.Razor.Language/src/DocumentationDescriptor.FormattedDescriptor.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Globalization;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.Razor.Language;
+
+internal abstract partial class DocumentationDescriptor
+{
+    private sealed class FormattedDescriptor : DocumentationDescriptor
+    {
+        public override object?[] Args { get; }
+
+        private string? _formattedString;
+
+        public FormattedDescriptor(DocumentationId id, object?[] args)
+            : base(id)
+        {
+#if DEBUG
+            foreach (var arg in args)
+            {
+                Debug.Assert(
+                    arg is string || arg is int || arg is bool || arg is null,
+                    "Only string, int, bool, or null arguments are allowed.");
+            }
+#endif
+
+            Args = args;
+        }
+
+        public override string GetText()
+            => _formattedString ??= string.Format(CultureInfo.CurrentCulture, GetDocumentationText(), Args);
```

Comment:
Is what I'm about to say incredibly unlikely? Yes. But this technically does have a race where we could return two different results for different calls if `CultureInfo.CurrentCulture` changes at the right time. I don't really think we need to _handle_ that, just pointing it out.

Reply:
Yup. You're absolutely correct - in all of the ways. :smile: